src/Func.cpp

```diff
@@ -1536,7 +1541,15 @@ void reorder_vars(vector<Dim> &dims_old, const VarOrRVar *vars, size_t size, con
 }
 
 Stage &Stage::reorder(const std::vector<VarOrRVar>& vars) {
-    reorder_vars(definition.schedule().dims(), &vars[0], vars.size(), *this);
+    const string &func_name = function.name();
+    vector<Expr> &args = definition.args();
+    vector<Expr> &values = definition.values();
+
+    // Check whether the operator is associative
+    const auto &prover_result = prove_associativity(func_name, args, values);
```

Comment:
Reordering also changes the order in which the reductions happen, as well as reassociating them, so I think you additionally need commutative.